src/Credentials/CredentialProvider.php

```diff
@@ -78,7 +79,10 @@ public static function defaultProvider(array $config = [])
         return self::memoize(
             self::chain(
                 self::env(),
+                self::assumeRole(),
```

Comment:
I thought about that, I didn't pass in the `$config` array because, I'd like to have the `defaultProvider()` plays the role of our SDK default credential chain instead of a provider where any customization could happen. I think it's a general good practice for our customer to be aware of what credential provider they are using when they really wants to apply some customization. Thoughts?

The current `$config` living there is serving for those instance credential providers.